packages/at_secondary_server/lib/src/verb/handler/notify_verb_handler.dart

```diff
@@ -57,260 +65,163 @@ class NotifyVerbHandler extends AbstractVerbHandler {
-              .build();
-          cachedKeyCommitId =
-              await _storeCachedKeys(key, metadata, atValue: atValue);
-          //write the latest commit id to the StatsNotificationService
-          _writeStats(cachedKeyCommitId, operation);
-          response.data = 'data:success';
-          return;
-        }
-
-        // Update metadata only if key is cached.
-        if (isKeyPresent) {
-          var atMetaData = AtMetadataBuilder(
-                  newAtMetaData: atMetadata,
-                  ttl: ttlMillis,
-                  ttb: ttbMillis,
-                  ttr: ttrMillis,
-                  ccd: isCascade,
-                  isEncrypted: isEncrypted)
-              .build();
-          cachedKeyCommitId = await _updateMetadata(notifyKey, atMetaData);
-          //write the latest commit id to the StatsNotificationService
-          _writeStats(cachedKeyCommitId, operation);
-          response.data = 'data:success';
-          return;
-        }
-        response.data = 'data:success';
+      // When connection is polAuthenticated, it indicates the receiver side of the
+      // the notification. Store the notification to the keystore.
+      else if (atConnectionMetadata.isPolAuthenticated) {
+        await _handlePolAuthenticatedConnection(
+            verbParams, atConnectionMetadata, response);
       }
     } finally {
       processNotificationMutex.release();
     }
   }
 
+  Future<void> _handlePolAuthenticatedConnection(
+      HashMap<String, String?> verbParams,
+      InboundConnectionMetadata atConnectionMetadata,
+      Response response) async {
+    logger.info('Storing the notification ${verbParams[AT_KEY]}');
+    var atNotificationBuilder = _populateNotificationBuilder(verbParams,
+        fromAtSign: atConnectionMetadata.fromAtSign!);
+    // If messageType is key, atMetadata is set in "_populateNotificationBuilder"
+    // When messageType is text, atNotificationBuilder.atMetadata fields are
+    // not applicable except "atMetadata.isEncrypted".
+    // So "atNotificationBuilder.atMetadata" will be overwritten
+    // "atMetadata.isEncrypted" represents if the message is encrypted or not.
+    if (atNotificationBuilder.messageType == MessageType.text) {
+      atNotificationBuilder.atMetaData = _atMetadataPool[
+          SecondaryUtil.getBoolFromString(verbParams[IS_ENCRYPTED])];
+    }
+    // Store the notification to the notification keystore.
+    await NotificationUtil.storeNotification(atNotificationBuilder.build());
+    OperationType operationType = getOperationType(verbParams[OPERATION]);
+    // When Operation is update, cache key only when TTR is set.
+    // So if TTR is null,  do nothing.
+    // Also, If operation is delete removed the cached key - irrespective of TTR value.
+    // So, If operation is not delete and TTR is null, return.
+    if (operationType != OperationType.delete &&
+        (_getTimeToRefresh(verbParams[AT_TTR]) == null)) {
```

Comment:
a value of 0 is equivalent to a value of null for ttr, please check for null or zero

Reply:
The method internally calls "validateTTR" in atmetadata_utils.dart in at_utils package which returns null when ttr is either "null" or "0"

```dart
if (ttr_ms == null || ttr_ms == 0) {
      return null;
    }
```